### PR TITLE
Add UI to view and download usage records

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR adds a new dashboard for viewing and downloading usage records.

This PR also makes `startdate` and `enddate` as non required params for `generateUsageRecords`. (Fixes: #7133)

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

### Screenshots (if appropriate):
image
image
image
image
image

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9 keepEnv

File: ui/src/components/view/ListView.vue [hunk 1]
```diff
@@ -164,7 +167,7 @@
         <router-link :to="{ path: createPathBasedOnVmType(record.vmtype, record.virtualmachineid) }">{{ text }}</router-link>
       </template>
       <template v-if="column.key === 'virtualmachinename'">
-        <router-link :to="{ path: getVmRouteUsingType(record) + record.virtualmachineid }">{{ text }}</router-link>
+        <router-link :to="{ path: getVmRouteUsingType(record) + record.virtualmachineid }">{{ text ? text : record.name }}</router-link>
```
Comment (shwstppr) [inline ui/src/components/view/ListView.vue:170]: should this be `record.virtualmachineid`?

File: ui/src/config/router.js [hunk 2]
```diff
@@ -221,9 +221,17 @@ export function asyncRouterMap () {
       generateRouterMap(offering),
       generateRouterMap(config),
       generateRouterMap(tools),
+      {
```
Comment (shwstppr) [inline ui/src/config/router.js:224]: do we need it defined here or can it be in an existing section or create a new section (already see infra/usage.js)?

File: ui/src/config/section/infra/usage.js [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+  }, {
+    name: 'comments',
+    component: shallowRef(defineAsyncComponent(() => import('@/components/view/AnnotationsTab.vue')))
+  }],
+  related: [{
+    name: 'vm',
+    title: 'label.instances',
+    param: 'hostid'
+  }],
+  actions: [
```
Comment (shwstppr) [inline ui/src/config/section/infra/usage.js:56]: Are the actions here actually used?

File: ui/src/views/infra/UsageRecords.vue [hunk 4]
```diff
@@ -0,0 +1,712 @@
...
+    </a-col>
+    <a-col :span="4">
+      <a-card
+        class="server-metrics"
+        :loading="serverMetricsLoading"
+      >
+        <a-statistic
+          :title="$t('label.status')"
+          :value="serverStats.state"
+          :valueStyle="{ color: serverStats.state === 'UP' ? 'green' : 'red' }"
```
Comment (shwstppr) [inline ui/src/views/infra/UsageRecords.vue:39]: Please check if there is any predefined values for color in config other wise good

**Review by shwstppr — COMMENTED** [review]
Great work @vishesh92!

Some initial comments

File: ui/src/config/section/infra/usage.js (hunk 3 above)
Comment (vishesh92) [reply ui/src/config/section/infra/usage.js:56]: This file is not being used. I was trying something else out and committed this file by mistake,

**Comment by DaanHoogland** [comment]
@vishesh92 it seems to me this menu item should be under "tools". what do you think?

**Comment by vishesh92** [comment]
> @vishesh92 it seems to me this menu item should be under "tools". what do you think?

IMO, Tools should have something which contains tools which are used very infrequently and Usage is something someone might use on a regular basis.
I don't have any strong opinions about this, but as of now tools have just **Comments** & **Import-Export Instances**. Let me know how should we proceed on this.

image

**Review by shwstppr — APPROVED** [review]
Code LGTM.

Others can comment on the placement of the menu option but I would prefer it under Tools

**Comment by DaanHoogland** [comment]
> Others can comment on the placement of the menu option but I would prefer it under Tools

metoo

**Comment by vishesh92** [comment]
@shwstppr @DaanHoogland Moved the menu item to under `Tools`.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan alma9 kvm-alma9 keepEnv

**Comment by DaanHoogland** [comment]
@blueorangutan alma9 kvm-alma9 keepEnv

**Comment by DaanHoogland** [comment]
> @blueorangutan alma9 kvm-alma9 keepEnv

build started manually, no idea why the bot refuses yet...

**Comment by DaanHoogland** [comment]
@blueorangutan alma9 kvm-alma9 keepEnv

**Review by rohityadavcloud — APPROVED** [review]
LGTM

**Comment by rohityadavcloud** [comment]
@blueorangutan ui

**Comment by vishesh92** [comment]
@blueorangutan ui

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test matrix

**Comment by borisstoyanov** [comment]
@blueorangutan test matrix

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9 keepEnv

**Review by bernardodemarco — COMMENTED** [review]
@vishesh92, interesting work. Here are the tests descriptions that I've done in a local environment and some small observations/suggestions.

## Tests Descriptions

<details>
    <summary>Usage records for 2024-07-15</summary>

image
</details>

---

<details>
    <summary>IP addresses usage records for the admin account on 2024-07-15</summary>

image
</details>

---

<details>
    <summary>IP addresses usage records for <code>b14d8275-fee9-4cd2-9fee-46c3da5cef78</code> resource on 2024-07-15</summary>

image
</details>

---

I exported the IP addresses usage records for the resource with ID `b14d8275-fee9-4cd2-9fee-46c3da5cef78` on 2024-07-15 into a CSV file. I then checked that the columns and their values are consistent with what is rendered on the ACS UI.

---

I filtered for a date interval in which no usage records were registered and confirmed that the CSV download operation correctly indicated that no usage records were found.

---

I created a `ROOT`'s child domain called `child-domain-1` with a domain admin account named `child-domain-1`. With this account, I acquired some public IPs and started a VM. After the usage job finished, I filtered for usage records with the `Fetch usage records for child domains` flag set to `true`.

I verified that the usage records of the accounts of the `ROOT` and `child-domain-1` domains are fetched correctly.

<details>
<summary>Fetch usage records for child domains</summary>

image
</details>

---

I also filtered for only the `child-domain-1` account usage records and confirmed they were fetched correclty.

<details>
<summary><code>child-domain-1</code> usage records</summary>

image
</details>

---

All previous tests were performed using UTC timezone. I then changed the UI setting to use the local timezone and started filtering records by date. I identified a small issue where the start and end date specified in the filter are not being converted correctly when calling the API.

For example, in the following screenshot, I'm filtering usage records from `2024-07-14`. However, the response includes records from `2024-07-13`.

image

---

There are also some minor enhacements that could improve the user experience, such as:
- Adding a button to copy the resource IDs, similar to the ones already displayed in the Info cards.
- Centering the filtering icon.
  - image

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by vishesh92** [comment]
@blueorangutan ui

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan ui

**Review by borisstoyanov — APPROVED** [review]
LGTM

**Comment by rohityadavcloud** [comment]
Merging based on review and manual QA. LGTM.

**Pull request merged**